src/components/__tests__/PropsTable.test.tsx

```diff
@@ -0,0 +1,309 @@
+import { render, screen, within } from '@testing-library/react'
+import { PropsTable, type ComponentProp } from '../PropsTable'
+
+const componentName = 'TestComponent'
+const componentDescription =
+  'This is the testable component for the PropsTable.'
+const propWithAllTableColumns: ComponentProp = {
+  name: 'propWithAllTableColumns',
+  type: 'string | () => void',
+  defaultValue: 'foobarbaz',
+  description: 'This prop has data for all table fields.',
+}
+const propWithNameOnly: ComponentProp = {
+  name: 'propWithNameOnly',
+}
+
+it('Renders component name in a heading level 3 by default', () => {
+  render(<PropsTable componentName={componentName} />)
+
+  expect(screen.getByRole('heading', { level: 3 })).toHaveTextContent(
+    componentName,
+  )
+})
+
+it('Renders component name in heading level when headingLevel is passed in', () => {
+  render(<PropsTable headingLevel="h2" componentName={componentName} />)
+
+  expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
+    componentName,
+  )
+})
+
+it('Does not render component description by default', () => {
+  render(<PropsTable componentName={componentName} />)
+
+  expect(screen.queryByTestId('component-description')).not.toBeInTheDocument()
```

Comment:
Feel like this happens every once in a blue moon, but only reason I'm doing this is because I feel like checking the specific description string doesn't exist could potentially lead to false passes. This at least checks the description wrapper (and its content) doesn't exist at all.